src/jit/optcse.cpp

```diff
@@ -748,6 +765,102 @@ unsigned Compiler::optValnumCSE_Locate()
     return 1;
 }
 
+//------------------------------------------------------------------------
+// updateCseArrLenMap: Check if this compare is a tractable function of
+//                     an array length that is a CSE candidate, and insert
+//                     an entry in the cseArrLenMap if so.  This facilitates
+//                     subsequently updating the compare's value number if
+//                     the array length gets CSEd.
+//
+// Arguments:
+//    compare - The compare node to check
+
+void Compiler::updateCseArrLenMap(GenTreePtr compare)
+{
+    assert(compare->OperIsCompare());
+
+    ValueNum  compareVN = compare->gtVNPair.GetConservative();
+    VNFuncApp cmpVNFuncApp;
+
+    if (!vnStore->GetVNFunc(compareVN, &cmpVNFuncApp) ||
+        (cmpVNFuncApp.m_func != GetVNFuncForOper(compare->OperGet(), (compare->gtFlags & GTF_UNSIGNED) != 0)))
```

Comment:
`(compare->gtFlags & GTF_UNSIGNED) != 0)`

Side note: maybe we should add `bool gtUnsigned()` to `GenTree`, this stuff is nightmarish. Too bad we don't have unsigned relops.